specutils/io/default_loaders/wcs_fits.py

```diff
@@ -86,3 +91,406 @@ def wcs1d_fits_writer(spectrum, file_name, **kwargs):
+                              'standard'.format(wcs_dict['ftype']))
+    else:
+        raise SyntaxError('dtype {:d} is not defined in the '
+                          'standard'.format(wcs_dict['dtype']))
+
+
+def _none():
+    """Required to handle No-wavelength solution
+    No wavelength solution is considered in the FITS standard (dtype = -1)
+    This method is placed here for completeness even though is not
+    implemented.
+    Raises:
+        NotImplementedError
+    """
+    raise NotImplementedError
+
+
+def _linear_solution(wcs_dict):
+    """Constructs a Linear1D model based on the WCS information obtained
+    from the header.
+    """
+    intercept = wcs_dict['crval'] - \
+                (wcs_dict['crpix'] - 1) * \
+                wcs_dict['cdelt']
+    model = models.Linear1D(slope=wcs_dict['cdelt'],
+                            intercept=intercept)
+
+    return model
+
+
+def _log_linear(wcs_dict):
+    """Not implemented
+    Raises:
+        NotImplementedError
+    """
+    # intercept = np.power(10, wcs_dict['crval'] +
+    #                          wcs_dict['cdelt'] *
+    #                         (wcs_dict['crpix'] - 1))
+    #
+    # slope = np.power(10, wcs_dict['cdelt'])
+    #
+    # model = models.Linear1D(slope=slope,
+    #                              intercept=intercept)
+    # print(model)
+    raise NotImplementedError
+
+
+def _chebyshev(wcs_dict):
+    """Returns a chebyshev model
+
+    Constructs a Chebyshev1D mathematical model
+
+    Parameters
+    ----------
+
+    wcs_dict : dict
+        Dictionary containing all the wcs information decoded from the header and
+        necessary for constructing the Chebyshev1D model.
+
+    Returns:
+
+    `~astropy.modeling.Model`
+
+    """
+    model = models.Chebyshev1D(degree=wcs_dict['order'] - 1,
+                               domain=[wcs_dict['pmin'],
+                                       wcs_dict['pmax']], )
+    # model.parameters[0] = wcs_dict['pmin']
+    for param_index in range(wcs_dict['order']):
+        model.parameters[param_index] = wcs_dict['fpar'][
+            param_index]
+
+    return model
+
+
+def _non_linear_legendre(wcs_dict):
+    """Returns a legendre model
+
+    Constructs a Legendre1D mathematical model
+
+    Parameters
+    ----------
+
+    wcs_dict : dict
+        Dictionary containing all the wcs information decoded from the header and
+        necessary for constructing the Legendre1D model.
+
+    Returns:
+
+    `~astropy.modeling.Model`
+
+    """
+    """Returns a legendre model"""
+    model = models.Legendre1D(degree=wcs_dict['order'] - 1,
+                              domain=[wcs_dict['pmin'],
+                                      wcs_dict['pmax']], )
+    # model.parameters[0] = wcs_dict['pmin']
+    for param_index in range(wcs_dict['order']):
+        model.parameters[param_index] = wcs_dict['fpar'][
+            param_index]
+
+    return model
+
+
+def _non_linear_lspline(wcs_dict):
```

Comment:
Add a simple description of what this _should_ do in the future.

Reply:
Done